lib/queue.js

```diff
@@ -680,6 +686,7 @@ Queue.prototype.processJob = function(job){
     _this.processing--;
     var error = err.cause || err; //Handle explicit rejection
     return job.moveToFailed(err)
+      .then(clearTimer)
```

Comment:
don't we have an hazard here where the timer is cleaned *after* we moved to failed?